File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_3_cross_cutting_concerns/before/StaticFileLogger.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_3_cross_cutting_concerns.before;

import com.grokkingsoftwarearchitecture.chapter04.shared.LogManager;

/**
 * ANTI-PATTERN: THE STATIC GOD.
 * ARCHITECTURE NOTE: Static utilities like this are global state.
 * They create "Hidden Dependencies" because they are called
 * internally without being declared in a constructor.
 */
public class StaticFileLogger {

    private StaticFileLogger() {
        // Private constructor to hide the implicit public one
    }

    public static void log(String message) {
        LogManager.info(StaticFileLogger.class, "(BEFORE_LOGGER) Static Log: {0}", message);
    }
}

File: Chapter 04/Python/section_4_3_cross_cutting_concerns/before/static_file_logger.py
"""
ANTI-PATTERN: MODULE-LEVEL STATIC LOGGING.
ARCHITECTURE NOTE: In Python, module-level functions often act 
as global state. While easy to use, they make unit testing 
difficult because they cannot be easily mocked or replaced.
"""
from shared.log_manager import LogManager
class StaticFileLogger:
    @staticmethod
    def log(message: str):
        LogManager.info("StaticFileLogger", "(BEFORE_LOGGER) Static Log: {0}", message)